### PR TITLE
NXP-22082 Copy-ResetSchema: Allow to not save the document

### PR DESCRIPTION
**Pull request by ThibArg**

### CODE REVIEW

File: nuxeo-features/nuxeo-automation/nuxeo-automation-core/src/main/java/org/nuxeo/ecm/automation/core/operations/document/CopySchema.java [hunk 1]
```diff
@@ -80,9 +83,11 @@ public DocumentModel run(DocumentModel target) throws OperationException {
         DocumentModel source = getDocumentFromIdOrPath();
         copySchemaProperties(source, target);
 
-        target = session.saveDocument(target);
-        if (save) {
-            session.save();
+        if(saveDocument) {
+        	target = session.saveDocument(target);
```
Comment (efge) [inline nuxeo-features/nuxeo-automation/nuxeo-automation-core/src/main/java/org/nuxeo/ecm/automation/core/operations/document/CopySchema.java]: Don't use tabs but spaces for indentation before `target = ...`.

File: nuxeo-features/nuxeo-automation/nuxeo-automation-core/src/main/java/org/nuxeo/ecm/automation/core/operations/document/CopySchema.java [hunk 2]
```diff
@@ -80,9 +83,11 @@ public DocumentModel run(DocumentModel target) throws OperationException {
         DocumentModel source = getDocumentFromIdOrPath();
         copySchemaProperties(source, target);
 
-        target = session.saveDocument(target);
-        if (save) {
-            session.save();
+        if(saveDocument) {
```
Comment (efge) [inline nuxeo-features/nuxeo-automation/nuxeo-automation-core/src/main/java/org/nuxeo/ecm/automation/core/operations/document/CopySchema.java]: Add a space between `if` and `(`.

File: nuxeo-features/nuxeo-automation/nuxeo-automation-core/src/main/java/org/nuxeo/ecm/automation/core/operations/document/ResetSchema.java [hunk 3]
```diff
@@ -72,9 +76,11 @@ private void resetSchemaProperties(DocumentModel target) throws OperationExcepti
     @OperationMethod(collector = DocumentModelCollector.class)
     public DocumentModel run(DocumentModel target) throws OperationException {
         resetSchemaProperties(target);
-        target = session.saveDocument(target);
-        if (save) {
-            session.save();
+        if(saveDocument) {
+        	target = session.saveDocument(target);
```
Comment (efge) [inline nuxeo-features/nuxeo-automation/nuxeo-automation-core/src/main/java/org/nuxeo/ecm/automation/core/operations/document/ResetSchema.java]: Same indent/spacing problems.

File: nuxeo-features/nuxeo-automation/nuxeo-automation-core/src/test/java/org/nuxeo/ecm/automation/core/operations/document/CopySchemaTest.java [hunk 4]
```diff
@@ -242,5 +243,28 @@ public void testMultipleTargetsFullSchema() throws OperationException {
...
+        source = session.saveDocument(source);
+        
+        target1.setPropertyValue("common:icon-expanded", "icon-expanded-target1");
+        target1 = session.saveDocument(target1);
+        
+        OperationContext context = new OperationContext(session);
+        context.setInput(target1);
+        OperationChain chain = new OperationChain("testSaveParameters_1");
+        chain.add(CopySchema.ID).set("sourceId", source.getId()).set("schema", "common").set("saveDocument", false);
+        /*DocumentModel ignore = (DocumentModel)*/service.run(context, chain);
```
Comment (efge) [inline nuxeo-features/nuxeo-automation/nuxeo-automation-core/src/test/java/org/nuxeo/ecm/automation/core/operations/document/CopySchemaTest.java]: Remove the commented-out code.

File: nuxeo-features/nuxeo-automation/nuxeo-automation-core/src/test/java/org/nuxeo/ecm/automation/core/operations/document/ResetSchemaTest.java [hunk 5]
```diff
@@ -186,5 +188,26 @@ public void testMultipleTargetsFullSchema() throws OperationException {
...
+    public void shouldNotSaveTheDocument() throws Exception {
+    	
+        target1.setPropertyValue("common:icon-expanded", "icon-expanded-target1");
+        target1 = session.saveDocument(target1);
+        
+        OperationContext context = new OperationContext(session);
+        context.setInput(target1);
+        OperationChain chain = new OperationChain("testSouldNotSaveTheDocument");
+        chain.add(ResetSchema.ID).set("schema", "common").set("saveDocument", false);
+        /*DocumentModel ignore = (DocumentModel)*/service.run(context, chain);
```
Comment (efge) [inline nuxeo-features/nuxeo-automation/nuxeo-automation-core/src/test/java/org/nuxeo/ecm/automation/core/operations/document/ResetSchemaTest.java]: Remove the commented-out code.

**Comment by efge** [comment]
👍

**Comment by kevinleturc** [comment]
👍

**Pull request merged**